internal/datacoord/compaction_task_l0.go

```diff
@@ -316,19 +311,33 @@ func (t *l0CompactionTask) BuildCompactionRequest() (*datapb.CompactionPlan, err
 			Deltalogs:     segInfo.GetDeltalogs(),
 			IsSorted:      segInfo.GetIsSorted(),
 		})
+		segments = append(segments, segInfo)
 	}
 
-	sealedSegmentIDs, sealedSegBinlogs := t.selectSealedSegment()
-	if len(sealedSegmentIDs) == 0 {
+	sealedSegments, sealedSegBinlogs := t.selectSealedSegment()
+	if len(sealedSegments) == 0 {
 		// TODO fast finish l0 segment, just drop l0 segment
 		log.Info("l0Compaction available non-L0 Segments is empty ")
 		return nil, errors.Errorf("Selected zero L1/L2 segments for the position=%v", taskProto.GetPos())
 	}
 
+	segments = append(segments, sealedSegments...)
+	logIDRange, err := PreAllocateBinlogIDs(t.allocator, segments)
+	if err != nil {
+		return nil, err
+	}
+	plan.PreAllocatedLogIDs = logIDRange
+	beginLogID, _, err := t.allocator.AllocN(1)
```

Comment:
Why not use logIDRange.Begin,  it saves One RPC to rootcoord.

Reply:
To avoid duplication of beginLogID and logIDRange, and since the coordination has been merged, the RPC overhead is minimal.

Reply:
agree with XuanYang-cn